src/apps/chifra/pkg/abi/load_abi.go

```diff
@@ -0,0 +1,236 @@
+package abi
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/cache"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/config"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/file"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Where to find know ABI files
+var knownAbiSubdirectories = []string{
+	"known-000", "known-005", "known-010", "known-015",
+}
+
+type AbiInterfaceMap = map[string]*types.SimpleFunction
+
+// getMapKey turns encoding into lower case, which can be used
+// as a key in AbiInterfaceMap
+func getMapKey(encoding string) string {
+	return strings.ToLower(encoding)
```

Comment:
Can we consolidate this in the same way as the Address thing? Seems similar.

Reply:
Yes, I will do it before merging Address thing.